homeassistant/components/barry/sensor.py

```diff
@@ -0,0 +1,63 @@
+"""Platform for sensor integration."""
+from datetime import timedelta
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(hours=1)
+MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Barry sensor."""
+    barry_connection = hass.data.get(DOMAIN)
+    price_code = hass.data.get("price_code")
+    async_add_entities([BarrySensor(barry_connection, price_code)], True)
+
+
+class BarrySensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, barry_home, price_code):
+        """Initialize the sensor."""
+        self._barry_home = barry_home
+        self.price_code = price_code
+        self._state = None
+        self._name = "Barry"
+        self._device_state_attributes = {}
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._device_state_attributes
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"Electricity price {self._name}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def model(self):
+        """Return the model of the sensor."""
+        return "Price Sensor"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return "DKK/kWh"
+
```

Comment:
```suggestion
```

Reply:
We also need to display unit of the electicity based on the customer region, it will display the unit dynamically. However, for time being, it has been hard coded at the moment. So, we need to keep "unit_of_measurement" function as it is.

Reply:
We don't, as it is moved to a class attribute in another suggestion.

See: <https://developers.home-assistant.io/docs/core/entity#entity-class-or-instance-attributes>

Reply:
<p style="box-sizing: border-box; margin-bottom: 16px; margin-top: 0px !important; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Thanks. I replaced all property methods to class attributes as mentioned, but HA is not able to identify them and displayed<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 11.9px; background-color: var(--color-markdown-code-bg); border-radius: 6px; margin: 0px; padding: 0.2em 0.4em;">UNKNOWN</code><span> </span>state and blank in<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 11.9px; background-color: var(--color-markdown-code-bg); border-radius: 6px; margin: 0px; padding: 0.2em 0.4em;">ATTRIBUTES</code><span> </span>column. I made changes as follows:<br style="box-sizing: border-box;">Code:<br style="box-sizing: border-box;"><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 11.9px; background-color: var(--color-markdown-code-bg); border-radius: 6px; margin: 0px; padding: 0.2em 0.4em;">_attr_name = 'Electricity price Barry'</code><br style="box-sizing: border-box;"><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 11.9px; background-color: var(--color-markdown-code-bg); border-radius: 6px; margin: 0px; padding: 0.2em 0.4em;">_attr_unit_of_measurement = "DKK/kWh"</code><br style="box-sizing: border-box;"><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 11.9px; background-color: var(--color-markdown-code-bg); border-radius: 6px; margin: 0px; padding: 0.2em 0.4em;">_attr_state = None</code></p><p style="box-sizing: border-box; margin-bottom: 16px; margin-top: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">HA displayes the result as follows:</p>

Entity | State | Attibutes
-- | -- | --
sensor.unnamed_device | unknown |  

<p style="box-sizing: border-box; margin-bottom: 0px !important; margin-top: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Can you please tell me what went wrong here?</p>

Reply:
Probably I set them wrong initially. But now I set them as class attributes as attached in the image. But I am still getting the same result as mentioned in the previous comment.
Let me know if I am still doing something wrong. 
Thanks.
![Screenshot from 2021-07-22 15-26-29](https://user-images.githubusercontent.com/77223861/126621558-fb781639-e3b4-474b-bdf9-8aab082cc19c.png)

Reply:
Have you removed/cleaned up the property methods that these replace?

Reply:
Yes, I removed/cleaned up all property methods and the image I attached is the only code for class Barry. Nothing more is there.

Reply:
The code in the screenshot is pretty much what I expect to work, it is how it is used a lot (search the codebase).

I would implement this file as:

```py
"""Platform for sensor integration."""
from datetime import timedelta

from homeassistant.components.sensor import SensorEntity
from homeassistant.util import Throttle

from .const import DOMAIN, PRICE_CODE

SCAN_INTERVAL = timedelta(hours=1)
MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)


async def async_setup_entry(hass, entry, async_add_entities):
    """Set up the Barry sensor."""
    barry_connection = hass.data[DOMAIN]
    price_code = hass.data[PRICE_CODE]
    async_add_entities([BarrySensor(barry_connection, price_code)], True)


class BarrySensor(SensorEntity):
    """Representation of a Sensor."""

    _attr_unit_of_measurement = "DKK/kWh"

    def __init__(self, barry_home, price_code):
        """Initialize the sensor."""
        self._barry_home = barry_home
        self.price_code = price_code
        self._attr_name = "Electricity price Barry"

    @Throttle(MIN_TIME_BETWEEN_UPDATES)
    def update(self):
        """Fetch new state data for the sensor.

        This is the only method that should fetch new data for Home Assistant.
        """
        self._attr_state = self._barry_home.update_price_data(self.price_code)
```